docs/components/custom_components.md

```diff
@@ -0,0 +1,115 @@
+---
+layout: default
+title: Custom Components
+parent: Components
+permalink: /docs/Custom_Components
+---
+
+# Implementing a Custom Component
+
+The components we provide here may not fit your specific needs, and you'll need to write a custom component. Below is an example of how this could be done using React.
+
+## Scenario
+
+Let's say you want to use the Discovery's [Answer finding](https://medium.com/ibm-data-ai/finding-concise-answers-to-questions-in-enterprise-documents-53a865898dbd). Instead of returning _documents_, we want to get _answers_ from within those documents.
+
+## Implement a custom Answers component
+
+To make things easier, we're going to reuse the `discovery-search-app` example code -- specifically, within `src/App.js`. We'll keep using the search field, but create a custom component to show the top 3 answers.
+
+First, we need to update the query that is sent out when a user presses `Enter` in the search field. The value is set as the `natural_language_query` field of the request body. But we also need to enable _answers_, as described in the article. To do so, we can set the `overrideQueryParameters` property for `<DiscoverySearch>`:
+
+```diff
+   ) : isError ? (
+     <div>Unable to load Discovery projects. Please check your console for more details.</div>
+   ) : (
+-    <DiscoverySearch searchClient={searchClient} projectId={projectId}>
++    <DiscoverySearch
++      searchClient={searchClient}
++      projectId={projectId}
++      overrideQueryParameters={{
++        passages: {
++          enabled: true,
++          max_per_document: 3,
++          characters: 850,
++          fields: ['title', 'text'],
++          find_answers: true,
++          max_answers_per_document: 1
++        },
++        _return: ['document_passages', 'extracted_metadata']
++      }}
++    >
+       <AppView />
+     </DiscoverySearch>
+   );
+```
+
+The overall details are explained in the article, but I did want to point out a couple things. First, `fields` should be set to reflect the data in your project/collection. This property tells Discovery in which document fields to look for answers. Second, we're setting `_return` to a set of top-level fields in the document results. Only the fields listed will be returned, cutting down on the response size (especially since we're not listing `text` and `html`, which can be quite large).
```

Comment:
maybe to avoid using I, we could word it as:
, but here are a few things to keep in mind.